app/src/main/java/zechs/drive/stream/ui/files/FilesFragment.kt

```diff
@@ -196,6 +196,19 @@ class FilesFragment : BaseFragment() {
             findNavController().navigate(action)
         } else if (file.isVideoFile) {
             launchVideoPlayer(file)
+        } else if (file.isShortcut) {
+            if (file.shortcutDetails.targetMimeType == "application/vnd.google-apps.folder") {
```

Comment:
Instead of doing this create variables such as [here](https://github.com/itszechs/DriveStream/blob/c6e7ce85648b44027684056716671a4f240278e1/app/src/main/java/zechs/drive/stream/data/model/DriveFile.kt#L17-L19) for cleaner code
https://github.com/itszechs/DriveStream/blob/c6e7ce85648b44027684056716671a4f240278e1/app/src/main/java/zechs/drive/stream/data/model/DriveFile.kt#L17-L19

Reply:
Made suggested changes. Type safety is a bit messy I suppose, but should work fine due to the logic flow. I have never worked with kotlin or android before, feel free to suggest further changes.